### DISCUSSION TITLE
Use opentelemetry-java project and an agent with different versions (combination manual and auto instrumentation)

### DISCUSSION THREAD

**Discussion by SimoneGiusso** [open-telemetry/opentelemetry-java#5654] [Q&A] [answered]
According to the [doc](https://opentelemetry.io/docs/concepts/instrumentation/#:~:text=Note%2C%20that%20for%20most%20languages,granular%20observability%20into%20your%20code.) it is possible to use auto and manual instrumentation together. Are there some possibilities to have problems if we use different versions? 

Suppose I use `opentelemetry-java-instrumentation:1.28.0` for the agent and `opentelemetry-java:1.27.0` for the manual instrumentation (or viceversa). I suppose it's quite safe for minor version but what about major versions? In other words, there are some risks to use both mechanism auto and manual? 

Moreover, would everything work if I just manual instrumenting the application as described [here](https://opentelemetry.io/docs/instrumentation/java/manual/#traces), or extra steps are needed to integrate both mechanisms? For example, can I get the span created by the auto-instrumentation (`Span.current()`) with the Tracer object created manually?

**Comment by mateuszrzeszutek** [comment, marked as answer]
> According to the [doc](https://opentelemetry.io/docs/concepts/instrumentation/#:~:text=Note%2C%20that%20for%20most%20languages,granular%20observability%20into%20your%20code.) it is possible to use auto and manual instrumentation together. Are there some possibilities to have problems if we use different versions?

It is possible, as long as the API version <= javaagent version.
We're not aware of any current problems, if you find one please create an issue.

> I suppose it's quite safe for minor version but what about major versions? In other words, there are some risks to use both mechanism auto and manual?

There's only one major version (1.x) right now; but even when we finally release javaagent 2.x I believe it will still be possible to use `opentelemetry-api` 1.x with it.

> Moreover, would everything work if I just manual instrumenting the application as described [here](https://opentelemetry.io/docs/instrumentation/java/manual/#traces), or extra steps are needed to integrate both mechanisms? For example, can I get the span created by the auto-instrumentation (`Span.current()`) with the Tracer object created manually?

Pretty much the only thing you have to remember is that you have to use the `GlobalOpenTelemetry` provided by the agent. All APIs should work out of the box.

**Reply by SimoneGiusso** [reply]
And which name I should specify to get the same Tracer used by the java agent i.e. `GlobalOpenTelemetry.getTracer(instrumentationScopeName)` (instrumentationScopeName = ???). I see the name is used the `ComponentRegistry` to get the `Tracer`.

**Reply by mateuszrzeszutek** [reply]
`instrumentationScopeName` is just the name of the instrumentation library, it identifies the source of telemetry. You can set it to `main` or your application name, or whatever you want really. It is a concept that's way more important for instrumentation libraries, since it allows determining what software component produces a given piece of telemetry.
